k8s/logs_test.go

```diff
@@ -0,0 +1,91 @@
+//go:build integration
+// +build integration
+
+package k8s_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"knative.dev/func/k8s"
+)
+
+func TestGetPodLogs(t *testing.T) {
+	var err error
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*5)
+	t.Cleanup(cancel)
+	cliSet, err := k8s.NewKubernetesClientset()
+	if err != nil {
+		t.Fatal(err)
+	}
+	testingNS := "pod-logs-test-ns-" + rand.String(5)
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: testingNS,
+		},
+		Spec: corev1.NamespaceSpec{},
+	}
+	_, err = cliSet.CoreV1().Namespaces().Create(ctx, ns, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() {
+		cliSet.CoreV1().Namespaces().Delete(ctx, testingNS, metav1.DeleteOptions{})
+	})
+	t.Log("created namespace: ", testingNS)
+
+	testingPodName := "testing-pod"
+
+	testMsg := "Hello World!"
+
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        testingPodName,
+			Labels:      nil,
+			Annotations: nil,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:    testingPodName,
+					Image:   "alpine",
+					Command: []string{"echo", "-n", testMsg},
+				},
+			},
+			RestartPolicy: corev1.RestartPolicyNever,
+		},
+	}
+
+	pod, err = cliSet.CoreV1().Pods(testingNS).Create(ctx, pod, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Log("created pod: " + testingPodName)
+
+out:
+	for i := 0; i < 600; i++ {
```

Comment:
Is there not an kube API call to wait until the pod is ready?

Reply:
It is quite low level. I am aware only of watch interface which is not much better than this with respect to line of code.

Reply:
https://github.com/knative/func/blob/84741e24794e039e17dd2d8292bf37502f376c88/k8s/persistent_volumes.go#L168-L193
Is it any better?

Reply:
> Is it any better?

I would not be surprised if it's doing something very similar.

Reply:
>I would not be surprised if it's doing something very similar.

Nah I think it is smarter than that, like doing some long polling on server or something.